runtime/src/main/scala/akka/grpc/internal/ClientState.scala

```diff
@@ -94,25 +98,41 @@ final class ClientState(settings: GrpcClientSettings, channelFactory: GrpcClient
     }
   }
 
-  private def create(): InternalChannel = {
-    val internalChannel: InternalChannel = channelFactory(settings)
-    internalChannel.done.onComplete {
+  private def create(): Future[InternalChannel] =
+    Patterns.retry(
+      () => channelFactory(settings),
+      // TODO get from settings
+      1000,
+      400.millis,
+      // TODO remove case once we update Akka
+      mat.asInstanceOf[ActorMaterializer].system.scheduler,
+      mat.asInstanceOf[ActorMaterializer].system.dispatcher)
+
+  private def recreateOnFailure(done: Future[Done]): Unit =
+    done.onComplete {
       case Failure(_: ClientConnectionException | _: NoTargetException) =>
```

Comment:
Is NoTargetException needed any ore here? Won't that be caught by the outer retry as it is on creation?

Reply:
Ah, indeed that can be dropped there I think.